mithril-doc/src/markdown_formatter.rs

```diff
@@ -0,0 +1,439 @@
+
+        let labels = headers
+            .iter()
+            .map(|(label, _, _)| label.to_string())
+            .collect::<Vec<String>>();
+
+        format!("| {} |\n|{}|", labels.join(" | "), sublines.join("|"))
+    }
+}
+
+pub fn doc_markdown_with_config(cmd: &mut Command, struct_doc: Option<&StructDoc>) -> String {
+    // See: https://github1s.com/clap-rs/clap/blob/HEAD/clap_builder/src/builder/command.rs#L1989
+
+    fn format_parameters(cmd: &Command, struct_doc: Option<&StructDoc>) -> String {
+        if cmd
+            .get_arguments()
+            .filter(|arg| argument_to_document(arg))
+            .peekable()
+            .peek()
+            .is_some()
+        {
+            let mut command_parameters = extract_clap_info::extract_parameters(cmd);
+            if let Some(config_doc) = struct_doc {
+                if !config_doc.data.is_empty() {
+                    command_parameters = command_parameters.merge_struct_doc(config_doc);
+                }
+            }
+
+            let parameters_table = doc_config_to_markdown(&command_parameters);
+
+            let parameters_explanation = "\n\
+                The configuration parameters can be set in either of the following ways:\n\
+                \n\
+                1. In a configuration file, depending on the `--run-mode` parameter. If the runtime mode is `testnet`, the file is located in `./conf/testnet.json`.\n\
+                \n\
+                2. The value can be overridden by an environment variable with the parameter name in uppercase.\n\
+                ";
+            format!("{}\n{}", parameters_explanation, parameters_table)
+        } else {
+            "".to_string()
+        }
+    }
+
+    fn format_subcommand(cmd: &Command) -> String {
+        let sub_commands = &mut cmd.get_subcommands().peekable();
+        if sub_commands.peek().is_some() {
+            let subcommands_lines = sub_commands
+                .map(|command| {
+                    vec![
+                        format!("**{}**", command.get_name()),
+                        command.get_all_aliases().collect::<Vec<&str>>().join(","),
+                        command.get_about().map_or("".into(), StyledStr::to_string),
+                    ]
+                })
+                .collect::<Vec<Vec<String>>>();
+
+            markdown::format_table(
+                &["Subcommand", "Aliases", "Performed action"],
+                &subcommands_lines,
+            )
+        } else {
+            String::from("")
+        }
+    }
+
+    fn format_command(
+        cmd: &mut Command,
+        parent: Option<String>,
+        struct_doc: Option<&StructDoc>,
+        level: usize,
+    ) -> String {
+        // It's important to start by calling `render_help` because that built the command.
+        // The initialization add `help` command and default values for parameters.
+        // This is why the command parameter is a mutable reference.
```

Comment:
Replace by:
// The Command object need to be completed calling `render_help` function.
// Otherwise there is no parameter default values and the `help` command is absent.